src/test/java/teammates/test/pageobjects/InstructorFeedbackEditPage.java

```diff
@@ -529,58 +533,39 @@ public boolean verifyNewRankOptionsQuestionFormIsDisplayed() {
         WebElement contribForm = browser.driver.findElement(By.id("rankOptionsForm"));
         return contribForm.isDisplayed() && addNewQuestionButton.isDisplayed();
     }
+    
     public boolean verifyNewRankRecipientsQuestionFormIsDisplayed() {
         WebElement contribForm = browser.driver.findElement(By.id("rankRecipientsForm"));
         return contribForm.isDisplayed() && addNewQuestionButton.isDisplayed();
     }
-    
-    public boolean verifyEndDatesBeforeTodayAreDisabled() {
-        Calendar cal = Calendar.getInstance();
-        
-        // if today is the 1st day of the month, the datepicker cannot navigate to the previous month (yesterday) 
-        if (cal.get(Calendar.DATE) != 1) {
-            cal.add(Calendar.DATE, -1);       
 
-            // get month names
-            DateFormatSymbols dfs = new DateFormatSymbols();
-            String[] months = dfs.getMonths();        
+    public boolean isDatesOfPreviousCurrentAndNextMonthEnabled() {
 
-            // yesterday's day, month and year
-            String day = Integer.toString(cal.get(Calendar.DATE));
-            String month = months[cal.get(Calendar.MONTH)];
-            String year = Integer.toString(cal.get(Calendar.YEAR));
+        // Navigate to the previous month
+        browser.driver.findElement(By.className("ui-datepicker-prev")).click();
 
-            // go to previous month
-            while (!(getDatepickerMonth().equals(month) && getDatepickerYear().equals(year))) {
-                browser.driver.findElement(By.className("ui-datepicker-prev")).click();
-            }
+        // Check if the dates of previous, current and next month are enabled 
+        for (int i = 0; i < 3; i++) {
+
+            List<WebElement> dates = browser.driver.findElements(By.xpath("//div[@id='ui-datepicker-div']/table/tbody/tr/td"));
+
+            for (WebElement date : dates) {
 
-            // different months with the same day (e.g. first week or last week of the month) in the datepicker
-            List<WebElement> daysWithSameDayAsYesterday = browser.driver.findElements(By.xpath("//td[a[text() = \"" + day + "\"]]"));
-            
-            for (WebElement yesterday : daysWithSameDayAsYesterday) {
-                // yesterday is enabled
-                if (!yesterday.getAttribute("class").contains("ui-datepicker-unselectable ui-state-disabled")) {
+                boolean isDisabled = date.getAttribute("class").contains("ui-datepicker-unselectable ui-state-disabled");
+                boolean isFromOtherMonth = date.getAttribute("class").contains("ui-datepicker-other-month");
```

Comment:
The logic about why this needs to be used is what I am not sure about. If the goal is to allow the user to select dates from the past, then shouldn't dates in other months be selectable too? In that case, the check should be independent of whether or not the datepicker is on another month. 

I might have made a logic error here, but with this variable, it seems that there is no reason to even navigate to the previous, or next month because the test can never fail on those months.